### ISSUE TITLE
Uniformly magnetized sphere

### ISSUE THREAD

**Issue by notsimon** [fatiando/fatiando#406]
Hi,

In order to reproduce the test data of [this article](http://user.it.uu.se/~thosc112/pubpdf/wahlstromksg2013.pdf), I am trying to compute the B-field of a uniformly magnetized sphere. I'm expected something like the following:

<img width="663" alt="capture d ecran 2017-07-26 a 09 18 11" src="https://user-images.githubusercontent.com/614955/28609091-7ca21cd6-71e3-11e7-92a4-9cd5b6465b4c.png">

Here's the code I'm using, a slightly modified version of one of the example:

```python
model = [
    mesher.Sphere(x=0, y=0, z=0, radius=1,
                  props={'magnetization': utils.ang2vec(1, inc=0, dec=0)})
]

shape = (50, 50)
area = [-2, 2, -2, 2]
x, y, z = gridder.regular(area, shape, z=0)

field = np.array([sphere.bx(x, y, z, model),
                  sphere.by(x, y, z, model),
                  sphere.bz(x, y, z, model)])
```

**Disclaimer**: I'm a programmer, not a physicist ;)

The sphere does not seam uniformly magnetized – or my understanding of what it means is inaccurate – as I am getting the following vector field (after a non-linear normalization for aesthetic purpose):

![index](https://user-images.githubusercontent.com/614955/28609153-bd2d94a6-71e3-11e7-9373-c9d2fc1c56b1.png)

Any idea how the first plot could be reproduced using Fatiando ?

Thanks in advance !

**Comment by leouieda** [comment]
Hi @notsimon thanks for reporting this and taking the time to write such a detailed issue! We really appreciate it.

I've never tested that part of the code inside the sphere (as you can see from [the tests](https://github.com/fatiando/fatiando/blob/master/fatiando/gravmag/tests/test_sphere.py)). It's good to know that it doesn't currently work. 

The equations we use are the same as those for a single dipole because outside of the sphere their fields are equivalent. But inside we would have to change the equations appropriately. I don't know exactly where to find the equations. 

@birocoles do you have any thoughts on this?

**Comment by notsimon** [comment]
Hi @leouieda, thanks for your explanation !
I guess that's more an enhancement than a bug to fix then, as the field inside the magnetized object may not be used very often.

Thanks for this awesome library !

**Comment by leouieda** [comment]
@notsimon that's true. I updated your title to reflect this so that we can keep track of our todo.

Would you be interested in contributing this if we find the equations?

**Comment by notsimon** [comment]
Hi @leouieda, sorry for the late reply.
Turns out I can manage to do what I need without what's happening inside the sphere.
If I find some time to contribute, I will probably use it to add Cuda support instead ! :)
